src/approval/ApprovalController.ts

```diff
@@ -0,0 +1,381 @@
+import BaseController, { BaseConfig, BaseState } from '../BaseController';
+
+const { ethErrors } = require('eth-rpc-errors');
+const { nanoid } = require('nanoid');
+
+const DEFAULT_TYPE = Symbol('DEFAULT_APPROVAL_TYPE');
+const STORE_KEY = 'pendingApprovals';
+
+type ApprovalType = string | typeof DEFAULT_TYPE;
```

Comment:
I'm not completely sure what you're asking, so I'm just going to give the full context for `DEFAULT_TYPE`. I apologize if this is more information than necessary!

We introduce approval types so that we can easily differentiate different kinds of requests (in most cases, the type will just be the name of an RPC method) and handle them appropriately. For example, as this will be used in the extension permissions controller, `wallet_requestPermissions` will be a request type, and each origin can only have a single such request pending at any one time.

I made the decision to make the request type optional. The `DEFAULT_TYPE` is an implementation detail used for requests that _have no type_ as far as consumers of this controller are concerned. However, due to the data structures used, we still need some kind of value to key these types. I decided to use a symbol for this value, to make it impossible for consumers to instantiate it.

`ApprovalType` is an internal type, and it only appears in the signatures of private methods. If the consumer does not specify a type, the type defaults to `DEFAULT_TYPE`. Since `DEFAULT_TYPE` is not a string, but we don't want to accept any other Symbols, `typeof DEFAULT_TYPE` is what we're left with.

On review, `DEFAULT_TYPE` is a bit of a misnomer, and I'm going to rename it to `NO_TYPE`.

Does this answer your question?

Reply:
`DEFAULT_TYPE` renamed to `NO_TYPE` in: 6998b65